src/Cli/dotnet/SlnFileFactory.cs

```diff
@@ -19,87 +49,57 @@ public static string[] ListSolutionFilesInDirectory(string directory, bool inclu
-            }
-            else
-            {
-                return FromDirectory(fileOrDirectory, includeSolutionXmlFiles);
-            }
-        }
+            string solutionPath = GetSolutionFileFullPath(fileOrDirectory, includeSolutionFilterFiles, includeSolutionXmlFiles);
 
-        private static SolutionModel FromFile(string solutionPath)
-        {
-            SolutionModel slnFile = null;
-            try
+            if (solutionPath.HasExtension(".slnf"))
             {
-                ISolutionSerializer serializer = SolutionSerializers.GetSerializerByMoniker(solutionPath) ?? throw new GracefulException(
+                return CreateFromFilteredSolutionFile(solutionPath);
+            }
+            ISolutionSerializer serializer = SolutionSerializers.GetSerializerByMoniker(solutionPath) ?? throw new GracefulException(
                     CommonLocalizableStrings.CouldNotFindSolutionOrDirectory,
                     solutionPath);
 
-                slnFile = serializer.OpenAsync(solutionPath, CancellationToken.None).Result;
-            }
-            catch (SolutionException e)
-            {
-                throw new GracefulException(
-                    CommonLocalizableStrings.InvalidSolutionFormatString,
-                    solutionPath,
-                    e.Message);
-            }
-            return slnFile;
+            return serializer.OpenAsync(solutionPath, CancellationToken.None).Result;
         }
 
-        private static SolutionModel FromDirectory(string solutionDirectory, bool includeSolutionXmlFiles)
+        public static SolutionModel CreateFromFilteredSolutionFile(string filteredSolutionFile)
         {
-            DirectoryInfo dir;
-            try
-            {
-                dir = new DirectoryInfo(solutionDirectory);
-                if (!dir.Exists)
-                {
-                    throw new GracefulException(
-                        CommonLocalizableStrings.CouldNotFindSolutionOrDirectory,
-                        solutionDirectory);
-                }
-            }
-            catch (ArgumentException)
-            {
-                throw new GracefulException(
-                    CommonLocalizableStrings.CouldNotFindSolutionOrDirectory,
-                    solutionDirectory);
-            }
-
-            FileInfo[] files = [
-                ..dir.GetFiles("*.sln"),
-                ..(includeSolutionXmlFiles ? dir.GetFiles(".slnx") : [])
-             ];
+            JsonDocument jsonDocument = JsonDocument.Parse(File.ReadAllText(filteredSolutionFile));
+            JsonElement jsonElement = jsonDocument.RootElement;
+            JsonElement filteredSolutionJsonElement = jsonElement.GetProperty("solution");
+            string originalSolutionPath = filteredSolutionJsonElement.GetProperty("path").GetString();
+            string originalSolutionPathAbsolute = Path.GetFullPath(Path.GetDirectoryName(originalSolutionPath), filteredSolutionFile);
```

Comment:
I think you put the Path.GetDirectoryName on the wrong thing here:
```suggestion
            string originalSolutionPathAbsolute = Path.GetFullPath(originalSolutionPath, Path.GetDirectoryName(filteredSolutionFile));
```

Reply:
Oh you're right, thx!